include/nil/blueprint/parser.hpp

```diff
@@ -449,7 +458,8 @@ namespace nil {
                     }
                     case llvm::Instruction::ICmp: {
                         auto cmp_inst = llvm::cast<const llvm::ICmpInst>(inst);
-                        if (cmp_inst->getOperand(0)->getType()->isIntegerTy())
+                        if (cmp_inst->getOperand(0)->getType()->isIntegerTy()
+                            || cmp_inst->getOperand(0)->getType()->isFieldTy())
                             handle_int_cmp(cmp_inst, variables);
```

Comment:
Let's rename this method to `handle_scalar_cmp`, because it's not only for integer values now.